apps/chrome-devtools/src/app-devtools/theming-panel/common.ts

```diff
@@ -41,3 +42,106 @@ export const resolveVariable = (
   }
   return variableValue;
 };
+
+type Variant =
+  | '50'
+  | '100'
+  | '200'
+  | '300'
+  | '400'
+  | '500'
+  | '600'
+  | '700'
+  | '800'
+  | '900'
+  | 'A100'
+  | 'A200'
+  | 'A400'
+  | 'A700';
+
+const VARIANTS: Variant[] = [
+  '50',
+  '100',
+  '200',
+  '300',
+  '400',
+  '500',
+  '600',
+  '700',
+  '800',
+  '900',
+  'A100',
+  'A200',
+  'A400',
+  'A700'
+];
+
+export const DEFAULT_VARIANT: Variant = '500';
+
+/* eslint-disable @typescript-eslint/naming-convention */
+const SATURATION_VALUES: Record<Variant, number> = {
+  '50': 0.91,
+  '100': 0.98,
+  '200': 0.96,
+  '300': 0.95,
+  '400': 0.96,
+  '500': 1,
+  '600': 1,
+  '700': 0.99,
+  '800': 0.89,
+  '900': 0.86,
+  'A100': 1,
+  'A200': 1,
+  'A400': 1,
+  'A700': 1
+};
+
+const LIGHTNESS_VALUES: Record<Variant, number> = {
+  '50': 0.12,
+  '100': 0.3,
+  '200': 0.5,
+  '300': 0.7,
+  '400': 0.86,
+  '500': 1,
+  '600': 0.87,
+  '700': 0.66,
+  '800': 0.45,
+  '900': 0.16,
+  'A100': 0.76,
+  'A200': 0.64,
+  'A400': 0.49,
+  'A700': 0.44
+};
+
+/* eslint-enable @typescript-eslint/naming-convention */
+
+/**
+ * Returns palette colors from one color
+ * Logic inspired from https://github.com/Artik-Man/material-theme-creator
+ * @param colorHex
+ */
+export const getPaletteColors = (colorHex: string): Record<string, string> => {
```

Comment:
maybe it could be nice to expose this helper directly in the framework, not sure @o3r/styling or @o3r/design ?